### ISSUE TITLE
Is there a way to over-ride the filter function?

### ISSUE THREAD

**Issue by jabberwo** [AllenFang/react-bootstrap-table#1205]
Is there a way I can provide the function to be called for deciding if a row should be present or not for a given filter value?

I've looked through the customer filter example, doesn't seem like it does what I need.  I'm not talking about customizing the look of the filter (I want a simple select), but I want to allow multiple text values to pass, for example.  But the text values are not the displayed value but rather in the row  data (the same data I can get out when I do sorting).  Something along the lines of sortFuncExtraData might work...

thanks!
jab

**Comment by jabberwo** [comment]
Problem would be solved if it worked like the Array.filter where it passed in the row data when the customer filter is called here:

	            case _Const2.default.FILTER_TYPE.CUSTOM:
	              {
	                valid = _this4.filterCustom(targetVal, filterVal, filterObj[key].value);
	                break;
	              }

filterVal is undefined since its a custom filter (stepping throughthe online demo with the Yes No checkboxes).

**Comment by jabberwo** [comment]
Aaahhh!

Finally found out what filterValue{function(cell,row){}) in the TableHeaderColumn header could do for me!  By passing in the whole row data object I was able to get what I want.

No on to the next issue -- turn off sorting when filtering is on...

**Comment by AllenFang** [comment]
@jabberwo, sorry for lately reply, is https://github.com/AllenFang/react-bootstrap-table/issues/1202 similar with your issue? and any bug issues here :) ?

Thanks 👍

**Comment by jabberwo** [comment]
Hi Allen,
I wasn't using handleFilterData.  At least, I didn't know I could.  If I could have provided on a per-row bases some extra data for filtering, that could be pulled out of the per-row data (i.e. a "dataField") then that would have solved my problem.  But I don't see how that's possible based on what I read in the documentation.
What I finally did was to use the filterValue function, which gets the whole row's data, to return the string I wanted to be used as the filter value.  
Just to show off what I've been doing with your table, which is probably not what you expect for a "data" table, I'm including a standalone example of what I just demoed for some high-er ups here at work!
Hope you enjoy what I've done with your child!
-= chuck
 "When once you have tasted flight you will always walk the earth with your eyes turned skyward; for there you have been and there you will always be." Leonardo da Vinci 

    On Sunday, April 9, 2017 5:04 AM, Allen <notifications@github.com> wrote:
 

 @jabberwo, sorry for lately reply, is #1202 similar with your issue? and any bug issues here :) ?Thanks 👍 —
